applications/MPMApplication/custom_conditions/particle_based_conditions/mpm_particle_base_dirichlet_condition.cpp

```diff
@@ -66,6 +66,74 @@ void MPMParticleBaseDirichletCondition::FinalizeSolutionStep( const ProcessInfo&
     m_imposed_displacement = ZeroVector(3);
 }
 
+void MPMParticleBaseDirichletCondition::CalculateInterfaceContactForce(const ProcessInfo& rCurrentProcessInfo )
+{
+    const unsigned int number_of_nodes = GetGeometry().PointsNumber();
+
+    GeneralVariables Variables;
+    MPMShapeFunctionPointValues(Variables.N);
+
+    array_1d<double, 3 > mpc_force = ZeroVector(3);
+
+    for ( unsigned int i = 0; i < number_of_nodes; i++ )
+    {
+        if (Variables.N[i] > std::numeric_limits<double>::epsilon() )
```

Comment:
Assuming you want this branch taken for negative values as well.

```suggestion
        if (Variables.N[i])
```

##

The only non-negative `double` smaller than `std::numeric_limits<double>::epsilon()` is `0`.

Reply:
Currently we have only positive values for N but just in case other shape functions will be used in the future we can add the abs here.